taf/developer_tool.py

```diff
@@ -45,6 +45,127 @@
+            )
+            num_of_signatures += 1
+        if num_of_signatures == keys_num:
+            all_loaded = True
+    taf_repo.writeall()
+
+
+def _load_signing_keys(
+    taf_repo,
+    role,
+    keystore=None,
+    role_key_infos=None,
+    loaded_yubikeys=None,
+    scheme=DEFAULT_RSA_SIGNATURE_SCHEME,
+):
+    """
+    Load role's signing keys. Make sure that at least the threshold of keys was
+    loaded, but allow loading more keys (so that a metadata file can be signed
+    by all of the role's keys if the user wants that)
+    """
+    print(f"Loading signing keys of role {role}")
+    threshold = taf_repo.get_role_threshold(role)
+    signing_keys_num = len(taf_repo.get_role_keys(role))
+    all_loaded = False
+    num_of_signatures = 0
+    keys = []
+    # check if keystore exists and if there is a file corresponding to this role
+    load_from_keystore = False
+    if keystore is not None:
+        keystore = Path(keystore)
+        # names of keys are expected to be role or role + counter
+        counter = "" if signing_keys_num == 1 else "1"
+        if (keystore / f"{role}{counter}").is_file():
+            load_from_keystore = True
+
+    while not all_loaded and num_of_signatures < signing_keys_num:
+        if signing_keys_num == 1:
+            key_name = role
+        else:
+            key_name = f"{role}{num_of_signatures + 1}"
+        if load_from_keystore:
+            # if loading from keystore, load all keys
+            password = None
+            if role_key_infos is not None and role in role_key_infos:
+                passwords = role_key_infos[role].get("passwords")
+                if passwords is not None and len(passwords):
+                    password = passwords[num_of_signatures]
+            if password is None:
+                password = getpass(
+                    f"Enter {key_name} keystore file password and press ENTER"
+                )
+
+            key = load_role_key(keystore, role, password, scheme)
+            keys.append(key)
+        else:
+            if num_of_signatures >= threshold:
+                all_loaded = not (
+                    click.confirm(
+                        f"Threshold of {role} keys reached. Do you want to load more root keys?"
+                    )
+                )
+            if not all_loaded:
+                is_yubikey = None
+                if role_key_infos is not None and role in role_key_infos:
+                    is_yubikey = role_key_infos[role].get("yubikey")
+                if is_yubikey is None:
+                    is_yubikey = click.confirm(f"Load {key_name} from YubiKey?")
+                if is_yubikey:
+                    yk.yubikey_prompt(key_name, role, taf_repo, False, loaded_yubikeys)
+                else:
+                    pem = getpass(
+                        f"Enter {key_name} private key without its header and footer"
+                    )
+                    pem = _form_private_pem(pem)
```

Comment:
Does it make sense to loop until a valid PEM is entered? If I enter something like "abcde" I get following error `securesystemslib.exceptions.CryptoError: RSA (public, private) tuple cannot be generated from the encrypted PEM string: Could not deserialize key data.`

Reply:
Moved all this code to a new function where we loop until a valid PEM is entered and check if it's a valid key of the given role